resource/resource_registry.go

```diff
@@ -63,7 +63,7 @@ type (
 
 	// LinkAssocationConfig allows one resource to associate a specific association config
 	// to its own config. This is generally done by a specific resource (e.g. data capture of many components).
-	LinkAssocationConfig[ConfigT any] func(conf ConfigT, resAssociation interface{}) error
+	LinkAssocationConfig func(conf *Config, resAssociation AssociatedConfig) error
```

Comment:
A future improvement could be making this a generic method on resAssociation but doing so would add to the list of generics the config depends on and that change would hit a lot of files.  Can do this if there is interest but didn't want to assume

Reply:
FWIW your removal of the generic parameter here makes sense to me as you've turned the `resAssociation` type from `interface{}` to `AssociatedConfig`. If anything, I would prefer avoiding generics unless they're necessary.